include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -91,21 +135,29 @@ class fpga_policy : public device_policy<KernelName>
   public:
     static constexpr unsigned int unroll_factor = factor;
 
-    fpga_policy()
-        : base(sycl::queue(
-#    if _ONEDPL_FPGA_EMU
-              __dpl_sycl::__fpga_emulator_selector()
-#    else
-              __dpl_sycl::__fpga_selector()
-#    endif // _ONEDPL_FPGA_EMU
-              ))
-    {
-    }
-
+    fpga_policy() = default;
     template <unsigned int other_factor, typename OtherName>
     fpga_policy(const fpga_policy<other_factor, OtherName>& other) : base(other.queue()){};
     explicit fpga_policy(sycl::queue q) : base(q) {}
     explicit fpga_policy(sycl::device d) : base(d) {}
+
+    operator sycl::queue() const { return queue(); }
+    sycl::queue
+    queue() const
```

Comment:
Am I correct that this function ```sycl::queue queue()``` should be declared as ```virtual``` now in this and base class?
1) Let's assume we have created the ```policy``` instance of ```fpga_policy```
    After this PR this instance doesn't have created ```sycl::queue``` instance inside as it's lazy initialized.
2) In some code we save result of ```static_cast<device_policy&>(policy)``` into ```auto& d_policy```:
```C++
    fpga_policy policy;
    auto& d_policy = static_cast<device_policy&>(policy);
```
3) and after that we call ```queue()``` function from reference to the base class:
```C++
    fpga_policy policy;
    auto& d_policy = static_cast<device_policy&>(policy);
    auto q = d_policy.queue();                   // call device_policy::queue() !?
```
Looks like we will call ```queue()``` method of the base class and creation of ```sycl::queue``` instance will be executed incorrectly.

Reply:
Or we need some another approach to resolve this issue.

Reply:
May be able to apply something like https://github.com/oneapi-src/oneDPL/pull/1158
but this PR doesn't resolve the problem of broken lazy initialization in constructor of
```C++
    template <unsigned int other_factor, typename OtherName>
    fpga_policy(const fpga_policy<other_factor, OtherName>& other) : base(other.queue()){};
```
See my comment https://github.com/oneapi-src/oneDPL/pull/1154/files#r1318324154

Reply:
no

Reply:
I feel like that:
- If we do that, this is not the design intent we want
- If we don't do that - this is the API breaking change